cmd/nvidia-dra-controller/imex.go

```diff
@@ -99,29 +103,34 @@ func StartIMEXManager(ctx context.Context, config *Config) (*ImexManager, error)
 
 // manageResourceSlices reacts to added and removed IMEX domains and triggers the creation / removal of resource slices accordingly.
 func (m *ImexManager) manageResourceSlices(ctx context.Context, owner resourceslice.Owner, addedDomainsCh <-chan string, removedDomainsCh <-chan string) error {
-	driverResources := resourceslice.DriverResources{}
-	controller, err := resourceslice.StartController(ctx, m.clientset, DriverName, owner, &driverResources)
+	driverResources := &resourceslice.DriverResources{}
+	controller, err := resourceslice.StartController(ctx, m.clientset, DriverName, owner, driverResources)
 	if err != nil {
 		return fmt.Errorf("error starting resource slice controller: %w", err)
 	}
 
+	imexDomainOffsets := new(imexDomainOffsets)
 	m.waitGroup.Add(1)
 	go func() {
 		defer m.waitGroup.Done()
 		for {
 			select {
 			case addedDomain := <-addedDomainsCh:
+				offset, err := imexDomainOffsets.add(addedDomain, ResourceSliceImexChannelLimit, DriverImexChannelLimit)
+				if err != nil {
+					klog.Errorf("Error calculating channel offset for IMEX domain %s: %v", addedDomain, err)
+					return
+				}
 				klog.Infof("Adding channels for new IMEX domain: %v", addedDomain)
-				newDriverResources := DriverResources(driverResources).DeepCopy()
-				newDriverResources.Pools[addedDomain] = generateImexChannelPool(addedDomain, ImexChannelLimit)
-				controller.Update(&newDriverResources)
-				driverResources = newDriverResources
+				driverResources := driverResources.DeepCopy()
```

Comment:
this was a bug -- we actually want the following (i.e. no `:` because we want to overwrite the original):
```
driverResources = driverResources.DeepCopy()
```
It's fixed in my refactoring:
https://github.com/NVIDIA/k8s-dra-driver/pull/189